modules/openapi-generator/src/main/resources/swift5/libraries/urlsession/URLSessionImplementations.mustache

```diff
@@ -398,25 +407,20 @@ private var urlSessionStore = SynchronizedDictionary<String, URLSession>()
     }
 }
 
-private class SessionDelegate: NSObject, URLSessionDelegate, URLSessionDataDelegate {
-
-    var credential: URLCredential?
-
-    var taskDidReceiveChallenge: ((URLSession, URLSessionTask, URLAuthenticationChallenge) -> (URLSession.AuthChallengeDisposition, URLCredential?))?
-
+private class SessionDelegate: NSObject, URLSessionTaskDelegate {
```

Comment:
@DavidHorvath I think here it should be `URLSessionDelegate`, or am I wrong?
Before it was `URLSessionDelegate, URLSessionDataDelegate` but I think `URLSessionDelegate` should be enough?
I would like to have your feedback on this please.

Reply:
The reason I changed the delegate to `URLSessionTaskDelegate` is because we actually use it to handle the challenges
[urlSession(_:task:didReceive:completionHandler:)](https://developer.apple.com/documentation/foundation/urlsessiontaskdelegate/1411595-urlsession) 

`URLSessionTaskDelegate` inherits from `URLSessionDelegate` and if we don't implement the session level challenge handler [urlSession(_:didReceive:completionHandler:)](https://developer.apple.com/documentation/foundation/urlsessiondelegate/1409308-urlsession), according to the documentation
`If you do not implement this method, the session calls its delegate’s urlSession(_:task:didReceive:completionHandler:) method instead.`

Reply:
Thanks, looks good to me 👍

Reply:
We have to add `URLSessionTaskDelegate` otherwise this delegate won't be called. 
The reason it worked before is because `URLSessionDataDelegate` was used which is inherited from `URLSessionTaskDelegate` so the delegate was called